pkg/operator/status.go

```diff
@@ -309,6 +311,30 @@ func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Depl
 	} else if cr.Spec.ManagementState == operatorapiv1.Removed {
 		operatorDegraded.Message = "The registry is removed"
 		operatorDegraded.Reason = "Removed"
+	} else if deploy == nil {
+		operatorDegraded.Status = operatorapiv1.ConditionTrue
+		operatorDegraded.Message = "The deployment does not exist"
+		operatorDegraded.Reason = "DeploymentNotFound"
+	} else if operatorAvailable.Status != operatorapiv1.ConditionTrue {
+		updatedAvailableCondition := v1helpers.FindOperatorCondition(cr.Status.Conditions, operatorapiv1.OperatorStatusTypeAvailable)
+		if updatedAvailableCondition != nil && time.Since(updatedAvailableCondition.LastTransitionTime.Time) > time.Minute {
+			operatorDegraded.Status = operatorapiv1.ConditionTrue
+			operatorDegraded.Message = updatedAvailableCondition.Message
+			operatorDegraded.Reason = "Unavailable"
+		}
+	} else if !isDeploymentStatusComplete(deploy) {
+		for _, cond := range deploy.Status.Conditions {
+			if cond.Type != appsapi.DeploymentProgressing {
+				continue
+			}
+			if cond.Reason != "ProgressDeadlineExceeded" {
```

Comment:
If we want to rely on `progressDeadlineSeconds` of the registry deployment, I'd say we should be more opinionated and set it to 1 minute.

Reply:
Done, PTAL.